velox/functions/prestosql/MapFromEntries.cpp

```diff
@@ -107,15 +113,24 @@ class MapFromEntriesFunction : public exec::VectorFunction {
 
             // Set the sizes to 0 so that the final map vector generated is
             // valid in case we are inside a try. The map vector needs to be
-            // valid because its consumed by checkDuplicateKeys before try sets
-            // invalid rows to null.
+            // valid because its consumed by checkDuplicateKeys before try
+            // sets invalid rows to null.
             mutableSizes[row] = 0;
             VELOX_USER_FAIL("map entry cannot be null");
           }
 
-          const bool isMapKeyNull =
-              keyValueVector->isNullAt(decodedValueVector->index(offset + i));
-          VELOX_USER_CHECK(!isMapKeyNull, "map key cannot be null");
+          // Check null keys.
+          auto keyIndex = decodedRowVector->index(offset + i);
+          if (keyVector->isNullAt(keyIndex)) {
+            VELOX_USER_FAIL(kNullKeyErrorMessage);
```

Comment:
nit: Why not VELOX_USER_CHECK(!keyVector->isNullAt(keyIndex), kNullKeyErrorMessage) ?

Reply:
will update before landing